### PR TITLE
Added method to get default header for `AbstractArray`

### PR DESCRIPTION
**Pull request by siddharthlal25**
Added this method to get default header of an `AbstractArray`, it is being used in `CCDReduction.jl` to create `CCDData` directly from an `AbstractArray`. E. g.
```
CCDData(data::AbstractArray) = CCDData(data, get_default_header(data))
```

### CODE REVIEW

File: src/header.jl [hunk 1]
```diff
@@ -422,3 +422,33 @@ function show(io::IO, hdr::FITSHeader)
         i != n && println(io)
     end
 end
+
+"""
+    get_default_header(data::AbstractArray)
+
+Creates a default header from an `AbstractArray` with fields being `SIMPLE`, `BITPIX`, `NAXIS`, axis lengths and `EXTEND`.
```
Comment (mileslucas) [inline src/header.jl:429]:
```suggestion
Creates a default header for the given array with the `SIMPLE`, `BITPIX`, `NAXIS`, `NAXIS*`, and `EXTEND` entries.
```

File: src/header.jl [hunk 2]
```diff
@@ -422,3 +422,33 @@ function show(io::IO, hdr::FITSHeader)
...
+"""
+function get_default_header(data::AbstractArray{T}) where T <: Number
+    # assigning keys
+    hdu_keys = ["SIMPLE",
+                "BITPIX",
+                "NAXIS",
+                [string(Symbol("NAXIS", i)) for i in 1:ndims(data)]...,
```
Comment (mileslucas) [inline src/header.jl:436]: `["NAXIS$i" for i in 1:ndims(data)]...`

is there a way to accomplish this withou allocating this intermediate array? (does this even allocate the intermediate array?)

File: src/header.jl [hunk 3]
```diff
@@ -422,3 +422,33 @@ function show(io::IO, hdr::FITSHeader)
...
+                  FITSIO.Libcfitsio.bitpix_from_type(T),          # BITPIX
+                  ndims(data),                                    # NAXIS
+                  reverse(size(data))...,                         # size of each axis
+                  true]                                           # EXTEND
+
+    # assigning comments
+    comments = ["file does conform to FITS standard",                                   # comment for SIMPLE
+                "number of bits per data pixel",                                        # comment for BITPIX
+                "number of data axes",                                                  # comment for NAXIS
+                [string(Symbol("length of data axis ", i)) for i in 1:ndims(data)]...,  # comments for axis length
```
Comment (mileslucas) [inline src/header.jl:450]: same comment as above

File: src/header.jl (hunk 2 above)
Comment (siddharthlal25) [reply src/header.jl]: There were around 60 allocations on this version (`@benchmark get_default_header(randn(512, 512))`), the newer version comes down to 48.

**Comment by kbarbary** [comment]
`get_...` gives me the impression that the thing you're getting is already an attribute or property of the argument, but in this case, you're *creating* a header based on the image. I suggest just `default_header(data)`.

File: src/header.jl [hunk 4]
```diff
@@ -422,3 +422,33 @@ function show(io::IO, hdr::FITSHeader)
...
+    # assigning keys
+    hdu_keys = ["SIMPLE",
+                "BITPIX",
+                "NAXIS",
+                ("NAXIS$i" for i in 1:ndims(data))...,
+                "EXTEND"]
+
+    # assiging values
+    hdu_values = [true,                                           # SIMPLE
+                  Libcfitsio.bitpix_from_type(T),                 # BITPIX
```
Comment (giordano) [inline src/header.jl:441]: This module is deprecated now.  I guess this should be something like
```suggestion
                  CFITSIO.bitpix_from_type(T),                 # BITPIX
```

**Pull request merged**